ObjectMemory_Allocation.c

```diff
@@ -73,22 +73,22 @@ ObjectPointer ObjectMemory_attemptToAllocateChunkInCurrentSegment(Word size) {
    */
   if (size < BigSize) {
     objectPointer = ObjectMemory_removeFromFreeChunkList(size);
-    if (objectPointer) {
+    if (objectPointer != NilPointer) {
       return objectPointer; /*ST:  small chunk of exact size handy so use it */
     }
   }
 
   /* we got here, so recycling an earlier allocation didn't work. Try to find a large chunk. */
   predecessor = NonPointer; /*ST:  remember predecessor of chunk under consideration */
-  objectPointer = ObjectMemory_headOfFreeChunkList_inSegment(LastFreeChunkList, currentSegment);
+  objectPointer = ObjectMemory_headOfFreeChunkList_inSegment(BigSize, currentSegment);
   /*ST:  the search loop stops when the end of the linked list is encountered */
   while(objectPointer != NonPointer) {
     availableSize = ObjectMemory_sizeBitsOf(objectPointer);
     if (availableSize == size) { /*ST:  exact fit - remove from free chunk list and return */
       next = ObjectMemory_classBitsOf(objectPointer); /*ST:  the link to the next chunk */
       if (predecessor == NonPointer) {
         /*ST:  it was the head of the list; make the next item the head */
-        ObjectMemory_headOfFreeChunkList_inSegment_put(LastFreeChunkList, currentSegment, next);
+        ObjectMemory_headOfFreeChunkList_inSegment_put(BigSize, currentSegment, next);
```

Comment:
These are all bugs that were uncovered by `AllocateSmallObjectBySplittingLargeChunk`